lib/dependency_validator.dart

```diff
@@ -51,41 +52,47 @@ Future<Null> run() async {
       ?.toList();
   final excludes = configExcludes ?? <Glob>[];
   logger.fine('excludes:\n${bulletItems(excludes.map((g) => g.pattern))}\n');
-  final ignoredPackages = <String>[...commonBinaryPackages, ...config?.ignore ?? []];
+  final ignoredPackages = config?.ignore ?? <String>[];
   logger.fine('ignored packages:\n${bulletItems(ignoredPackages)}\n');
 
   // Read and parse the analysis_options.yaml in the current working directory.
   final optionsIncludePackage = getAnalysisOptionsIncludePackage();
 
   // Read and parse the pubspec.yaml in the current working directory.
-  final YamlMap pubspecYaml = loadYaml(File('pubspec.yaml').readAsStringSync());
+  final pubspec = Pubspec.parse(File('pubspec.yaml').readAsStringSync());
 
   // Extract the package name.
-  final packageName = pubspecYaml[nameKey];
+  final packageName = pubspec.name;
 
   logger.info('Validating dependencies for $packageName\n');
 
   // Find packages that provide executables.
   final packagesWithExecutables = Set<String>();
+  final packagesWithBuilders = Set<String>();
   final packageConfig = await findPackageConfig(Directory.current);
   for (final package in packageConfig.packages) {
+    // Search for builders
+    final buildConfig = await BuildConfig.fromBuildConfigDir(packageName, pubspec.dependencies.keys, package.root.path);
```

Comment:
Also, since this process is more involved than the executable detection (have to read and parse two yaml files), I do think we should only do this for packages that are explicit dependencies and not every package in the package config file.

Reply:
Oh oops I'm an idiot.